[Brendan Tan] iP

brendantwh opened the pull request:
# Duke

> I am Duke

I am Duke.

1. Download Duke [here](https://github.com/nerbnerb/ip/releases/latest)
2. Run `java -jar duke.jar` in your terminal wherever Duke.jar is located
3. Add your tasks ✅
4. Get distracted by how well Duke manages your tasks

And it is **FREE**[^1]!

### Features

- [x] Add and remove tasks
- [x] Include time of deadline and events
- [x] Mark your tasks as done (or not)
- [x] View your tasks
- [ ] Reminders (soon<sup>TM</sup>)

### Date/time syntax

You can omit the year to default to the current year, and the space between the time and AM/PM.
For example, to key in the date 23 August 2022, time 6:00pm, you can use:

- `d MMM yyyy h:mm a`
  - 23 Aug 2022 6:00 pm
- `d MMM yyyy h a`
  - 23 Aug 2022 6pm
- `d MMM h:mm a`
  - 23 Aug 6:00pm
- `d MMM h a`
  - 23 Aug 6 pm
- `d/M/yyyy HHmm`
  - 23/8/2022 1800
- `d/M HHmm`
  - 23/8 1800

---

Here is some Java code
```java
public static void main(String[] args) {
    Path dataPath = Paths.get("data", "duke.txt");
    
    new Duke(dataPath).run();
    
    DukeResponse.outro();
}
```

[^1]: While in development; thereafter $99.99/month.

Commit:
Add Gradle support

brendantwh committed:
Level 1. Greet, Echo, Exit

brendantwh committed:
Level 2. Add, List

brendantwh committed:
Level 3. Mark as Done

brendantwh committed:
Level 4. ToDos, Events, Deadlines

brendantwh committed:
Automated Text UI Testing

brendantwh committed:
Level 5. Handle Errors

brendantwh committed:
Fix error when not entering number for mark/unmark

brendantwh committed:
Level 6. Delete

brendantwh committed:
Use Enums

brendantwh committed:
Add ability to save and load task list

brendantwh committed:
Add date and time conversion from string

brendantwh committed:
Merge branch 'branch-Level-7'

brendantwh committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	text-ui-test/EXPECTED.TXT
#	text-ui-test/input.txt

brendantwh committed:
Modify classes to adopt more OOP

brendantwh committed:
Move classes into packages

brendantwh committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

brendantwh committed:
Add Gradle support

brendantwh committed:
Add JUnit tests

brendantwh committed:
Add JavaDoc comments

brendantwh committed:
Tweak code to comply with coding standard

brendantwh committed:
Add search functionality

brendantwh committed:
Merge branch 'branch-A-JavaDoc'

brendantwh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/DukeUi.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java

brendantwh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/DukeList.java
#	src/main/java/duke/Parser.java

brendantwh committed:
Fix JavaDocs formatting

brendantwh committed:
Fix response messages

brendantwh committed:
Revert README to original

ReaganTan00 commented on src/test/java/duke/task/TodoTest.java:
```diff
@@ -0,0 +1,25 @@
+        assertEquals("", todo.getOtherData());
+    }
+
+    @Test
+    public void toString_returnString() {
+        assertEquals("[T][ ] borrow book", todo.toString());
+    }
+}
```
As per first peer PR review, clean work and couldn't find any common coding standard violations.
Making any other comment would really just be nitpicking, so good job 👍 .

ReaganTan00 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,46 @@
+
+public class ParserTest {
+    @Test
+    public void getResponse_validInput_returnCorrectResponse() {
+        DukeList list = new DukeList();
+        String input = "event bookfair /at 27 aug 4.30pm";
+        try {
+            DukeResponse response = Parser.getResponse(list, input);
+            assertEquals(EventResponse.class, response.getClass());
+            assertFalse(response.isExit());
+        } catch (DukeException e) {
+            fail(e);
+        }
+    }
```
Really clean naming conventions that follow coding standards.
👍

ReaganTan00 reviewed (commented):
Honestly couldn't find any coding standard violations and will be glad to be proven wrong. 
Really clean work !

brendantwh committed:
Add a GUI to Duke

brendantwh committed:
Merge branch 'branch-Level-10'

Jo3LW commented on src/main/java/duke/response/DukeResponse.java:
```diff
@@ -0,0 +1,36 @@
+
+import duke.DukeException;
+
+/**
+ * A response by Duke.
+ */
```
I like that all the Coding practices were followed like class labels. Good job :)

Jo3LW reviewed (commented):
Overall very clean and organised code. Great job 👍

brendantwh committed:
Update main class

brendantwh committed:
Use assertions

brendantwh committed:
Improve code quality

brendantwh committed:
Merge pull request #1 from nerbnerb/branch-A-Assertions

Use assertions

brendantwh committed:
Merge pull request #2 from nerbnerb/branch-A-CodeQuality

Improve code quality

brendantwh committed:
Add ability to sort tasks

sortby accepts the following 4 criterias: time, type, desc, status.
* time sorts the Tasks in chronological order.
* type sorts the Tasks according to Task type.
* desc sorts the Tasks according to lexical order of description.
* status sorts the Tasks according to whether the Task is done or not.

Comparator getter was added to Task class so that Comparator
can access protected fields and to encapsulate the
implementation of comparing between Tasks from DukeList.

brendantwh committed:
Merge branch 'branch-C-Sort'

brendantwh committed:
Add user guide

brendantwh committed:
Update config, split home page into multiple pages

brendantwh committed:
Fix home page not showing

brendantwh committed:
Change syntax examples to table

brendantwh committed:
Improve the GUI

brendantwh committed:
Change to save after every task list modification

brendantwh committed:
Remove redundant files

brendantwh committed:
Update bottom bar of GUI

brendantwh committed:
Update user guide feature list
